docs/source/llms/tracing/index.rst

```diff
@@ -0,0 +1,615 @@
+By using the :py:func:`mlflow.start_span` context manager, you can group multiple traced functions under a single parent span, making it easier to understand 
+the relationships between different parts of your code.
+
+The context handler is recommended when you need to refine the scope of data capture for a given span. If your code is logically constructed such that 
+individual calls to services or models are contained within functions or methods, on the other hand, using the decorator approach is more straight-forward 
+and less complex.
+
+.. code-block:: python
+
+    import mlflow
+
+
+    @mlflow.trace
+    def first_func(x, y=2):
+        return x + y
+
+
+    @mlflow.trace
+    def second_func(a, b=3):
+        return a * b
+
+
+    def do_math(a, x, operation="add"):
+        # Use the fluent API context handler to create a new span
+        with mlflow.start_span(name="Math") as span:
+            # Specify the inputs and attributes that will be associated with the span
+            span.set_inputs({"a": a, "x": x})
+            span.set_attributes({"mode": operation})
+
+            # Both of these functions are decorated for tracing and will be associated
+            # as 'children' of the parent 'span' defined with the context handler
+            first = first_func(x)
+            second = second_func(a)
+
+            result = None
+
+            if operation == "add":
+                result = first + second
+            elif operation == "subtract":
+                result = first - second
+            else:
+                raise AttributeError(f"Unsupported Operation Mode: {operation}")
+
+            # Specify the output result to the span
+            span.set_outputs({"result": result})
+
+            return result
+
+When calling the ``do_math`` function, a trace will be generated that has the root span (parent) defined as the 
+context handler ``with mlflow.start_span():`` call. The ``first_func`` and ``second_func`` calls will be associated as child spans
+to this parent span due to the fact that they are both decorated functions (having ``@mlflow.trace`` decorated on the function definition). 
+
+Running the following code will generate a trace. 
+
+.. code-block:: python
+
+    do_math(8, 3, "add")
+
+This trace can be seen within the MLflow UI:
+
+.. figure:: ../../_static/images/llms/tracing/trace-view.png
+    :alt: Trace within the MLflow UI 
+    :width: 100%
+    :align: center
+
+
+
+Function wrapping
+#################
+
+Function wrapping provides a flexible way to add tracing to existing functions without modifying their definitions. This is particularly useful when 
+you want to add tracing to third-party functions or functions defined outside of your control. By wrapping an external function with :py:func:`mlflow.trace`, you can
+capture its inputs, outputs, and execution context.
+
+
+.. code-block:: python
+
+    import mlflow
+
+
+    def my_external_function(x, y):
+        return x + y
```

Comment:
Can we use an external function? In the `mlflow.trace` example, we use `math`:

https://github.com/mlflow/mlflow/blob/920dbf423ba35bb11a46a85a650ef0b05ab12095/mlflow/tracing/fluent.py#L117-L121

Reply:
Rewrote the entire example showing both ways of wrapping and using math lib